x-pack/plugins/security/server/routes/session_management/info.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SessionInfo } from '../../../common/types';
+import { RouteDefinitionParams } from '..';
+
+/**
+ * Defines routes required for the session info.
+ */
+export function defineSessionInfoRoutes({ router, logger, session }: RouteDefinitionParams) {
+  router.get(
+    {
+      path: '/internal/security/session',
+      validate: false,
+      // We have both authenticated and non-authenticated sessions.
```

Comment:
__note:__ this change is technically not needed right now, but it still can be called without having a valid session (e.g. with proper `Authorization` header) so we'd need to handle this case anyway. But I moved forward with the solution that feels more correct since we have both authenticated and non-authenticated sessions

Reply:
Interesting, looking...

Reply:
Okay, I agree, the solution you've suggested makes sense, thanks! I'm going to return 204 for empty sessions though, as empty object I'm returning right now is rather confusing anyway.

Reply:
Done in https://github.com/elastic/kibana/pull/68117/commits/9548184103a930e94e14679750eeb571923722c8

Reply:
Well, we missed one case when session doesn't exist for non-anonymous routes, but we shouldn't log user out - when user is authenticated via HTTP headers (e.g. when there is a proxy in front of Kibana or when reports are generated - as it turned out it's the reason for failing Reporting tests). In this case we're able to authenticate user, but don't create session. 

The reason why `SessionTimeout` UI app kind of works right now in this case is that it uses [`try/catch` that swallows all possible errors](https://github.com/elastic/kibana/blob/a81059b14baebcd60e499fd3eaef425f00e91163/x-pack/plugins/security/public/session/session_timeout.tsx#L107-L118), and in case of empty session the error is `Uncaught TypeError: sessionInfo.provider is undefined` :slightly_smiling_face: 

I think we need to take a step back and find a proper solution for all possible cases, but in this PR I'm going to just return "auth required" requirement for the session info route to preserve current behavior since nothing else is using it anyway.

What do you think?

Reply:
> The reason why `SessionTimeout` UI app kind of works right now in this case is that it uses [`try/catch` that swallows all possible errors](https://github.com/elastic/kibana/blob/a81059b14baebcd60e499fd3eaef425f00e91163/x-pack/plugins/security/public/session/session_timeout.tsx#L107-L118), and in case of empty session the error is `Uncaught TypeError: sessionInfo.provider is undefined` 🙂

😬

> I think we need to take a step back and find a proper solution for all possible cases, but in this PR I'm going to just return "auth required" requirement for the session info route to preserve current behavior since nothing else is using it anyway.
> 
> What do you think?

Agree. I'm happy to lend a hand with this in a follow-on PR.

Reply:
>  in this PR I'm going to just return "auth required" requirement for the session info route to preserve current behavior since nothing else is using it anyway.

++ makes sense to me too